go/saltpack/decrypt.go

```diff
@@ -145,26 +145,27 @@ func (ds *decryptStream) tryVisibleReceivers(hdr *EncryptionHeader, ephemeralKey
 
 	i, sk := ds.ring.LookupBoxSecretKey(kids)
 	if i < 0 || sk == nil {
-		return nil, nil, nil, -1, nil
+		return nil, nil, SymmetricKey{}, -1, nil
 	}
 
-	// Decrypt the sender's public key
 	shared := sk.Precompute(ephemeralKey)
 
 	orig, ok := tab[i]
 	if !ok {
-		return nil, nil, nil, -1, ErrBadLookup
+		return nil, nil, SymmetricKey{}, -1, ErrBadLookup
 	}
 
-	keysRaw, err := shared.Unbox(ds.nonce.ForKeyBox(), hdr.Receivers[orig].Keys)
+	payloadKeySlice, err := shared.Unbox(ds.nonce.ForKeyBox(), hdr.Receivers[orig].PayloadKeyBox)
 	if err != nil {
-		return nil, nil, nil, -1, err
+		return nil, nil, SymmetricKey{}, -1, err
 	}
+	var payloadKey SymmetricKey
```

Comment:
can you make a construct that loads from a slice?  it should potentially error out too